src/sentry/integrations/fake_log_integration/metric_alerting/__init__.py

```diff
@@ -0,0 +1,49 @@
+from sentry.constants import ObjectStatus
+from sentry.incidents.action_handlers import DefaultActionHandler
+from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
+from sentry.incidents.models.incident import IncidentStatus
+from sentry.integrations.fake_log_integration.log_provider import FakeIntegrationClient
+from sentry.integrations.services.integration import integration_service
+
+
+@AlertRuleTriggerAction.register_type(
+    "fake_log",
+    AlertRuleTriggerAction.Type.FAKE_LOG,
+    [AlertRuleTriggerAction.TargetType.SPECIFIC],
+)
+class FakeLogActionHandler(DefaultActionHandler):
+    @property
+    def provider(self) -> str:
+        return "fake_log"
+
+    def send_alert(
+        self,
+        metric_value: int | float,
+        new_status: IncidentStatus,
+        notification_uuid: str | None = None,
+    ):
+        # This pattern of retrieving the integration is common across alerting functions
+        result = integration_service.organization_context(
+            organization_id=self.incident.organization_id, integration_id=self.action.integration_id
+        )
+        integration = result.integration
+        org_integration = result.organization_integration
+        # This handling is also common, but not always consistent
+        if (
+            org_integration is None
+            or integration is None
+            or integration.status != ObjectStatus.ACTIVE
+        ):
+            # Integration removed, but rule is still active.
+            return False
+        identifier = self.action.target_identifier
```

Comment:
Hmm - in general, this feels like it's mixing specific / generic concepts in this portion to figure out the data to send. maybe we can split out the generic things as a separate / flat method then we only have handler specific code here

Reply:
I think so. Merging a lot of this integration-specific logic into service classes or the corresponding integration installations seems like a good initial milestone.